plugins/usi-dock/fu-usi-dock-dmc-device.c

```diff
@@ -55,6 +56,28 @@ fu_usi_dock_dmc_device_parent_notify_cb(FuDevice *device, GParamSpec *pspec, gpo
 			return;
 		}
 
+		/* allow matching PCB version */
+		serialnum = fu_device_get_serial(device);
+		if (strlen(serialnum) >= 8) {
```

Comment:
I think this is going to explode if the serial is ever NULL. You need `if (serialnum != NULL && strlen(serialnum) >= 8) {`